job_runner.py

```diff
@@ -0,0 +1,97 @@
+import os
+from jsonschema import validate, ValidationError
+import yaml
+from test_runner import TestRunner
+from compliance_suite.exceptions.ComplianceException import ComplianceException
+from compliance_suite.functions.colored_console import print_blue, print_red, print_green, print_yellow
+
+
+class JobRunner():
+
+    def __init__(self, tags):
+        self.path = os.getcwd()
+        self.tags = tags
+        self.test_count = 0
+        self.result = {             # To store the count of passed/failed/skipped tests
+            "passed": 0,
+            "failed": 0,
+            "skipped": 0
+        }
+        self.test_status = {        # To store the status of each test
+            "passed": [],
+            "failed": [],
+            "skipped": []
+        }
+
+    def generate_summary(self):
+
+        passed_tests = ", ".join(self.test_status["passed"])
+        failed_tests = ", ".join(self.test_status["failed"])
+        skipped_tests = ", ".join(self.test_status["skipped"])
+
+        print("\n\n\n")
+        print_yellow("{:#^90}".format("   Compliance Testing Summary   "))
+        print_yellow("#{:^88}#".format(""))
+        print_yellow("#{:^88}#".format(f"Total Tests - {self.test_count}"))
+        print_yellow("#{:^88}#".format(f'Passed - {self.result["passed"]} ({passed_tests})'))
+        print_yellow("#{:^88}#".format(f'Failed - {self.result["failed"]} ({failed_tests})'))
+        print_yellow("#{:^88}#".format(f'Skipped - {self.result["skipped"]} ({skipped_tests})'))
+        print_yellow("#{:^88}#".format(""))
+        print_yellow("{:#^90}".format(""))
+
+    def validate_job(self, yaml_data, yaml_file):
+        schema_path = os.path.join(self.path, "tests", "template", "test_template_schema.json")
+        with open(schema_path, "r") as f:
+            json_schema = yaml.safe_load(f)
+
+        try:
+            validate(yaml_data, json_schema)
+            print(f'Test YAML file valid for {yaml_file}')
+        except ValidationError as err:
+            raise ComplianceException(f"YAML schema validation error - {yaml_file}. Error details - {err.message}")
+
+    def validate_yaml(self, file, yaml_file):
+        try:
+            yaml_data = yaml.safe_load(file)
+            return yaml_data
+        except yaml.YAMLError as err:
+            raise ComplianceException(f"Invalid YAML File - {yaml_file}. Error details - {err}")
+
+    def run_jobs(self):
+
+        yaml_path = os.path.join(self.path, "tests")
```

Comment:
Better to parametrize this, i.e., pass the entire folder with tests as an argument to the constructor. Should take this from CLI.

Reply:
Passing as argument seems a better approach as it had become messy to handle the file paths. However, after we split the repo, we will extract the files from "tests" repo. Can we leave this for later as it will be updated then.

Reply:
Sure. Just add these things to the issue description (or a comment) so that they won't be forgotten